### PR TITLE
Attempted fix for Issue #145 - Male Mod Clothing Items Overwriting Female non-Mod Items

### PR DESCRIPTION
**Pull request by OrangeSpork**
Taking a crack at fixing this. This definitively resolves the issue for chara cards by (optionally) splitting the category lists into M and F (as well as a base Combined) versions and using the correct categories for the card.

For coordinate save/loads...the issue is that coordinate cards aren't marked by sex and there's no link from the coordinate to the enclosing character card. Barring any brighter ideas this saves the sex of the card into the extended data on the clothes when creating new coordinate cards so those will work correctly. Older coordinate cards will have to continue to use all categories and be vulnerable to the issue.

### CODE REVIEW

File: src/Core_Sideloader/UniversalAutoResolver/Core.UAR.StructReference.AI.cs [hunk 1]
```diff
@@ -279,34 +279,59 @@ internal static class StructReference
 #endregion
 
 #region ChaFileClothes
-        private static Dictionary<CategoryProperty, StructValue<int>> ChaFileClothesGenerator()
+        private static Dictionary<CategoryProperty, StructValue<int>> ChaFileClothesGenerator(bool male, bool female)
```
Comment (ManlyMarco) [inline src/Core_Sideloader/UniversalAutoResolver/Core.UAR.StructReference.AI.cs:282]: Can both be true or false at the same time?

File: src/Core_Sideloader/UniversalAutoResolver/Core.UAR.Hooks.cs [hunk 2]
```diff
@@ -257,7 +258,11 @@ internal static void ExtendedCoordinateLoad(ChaFileCoordinate file)
                     }
                 }
 
+#if AI || HS2                
+                IterateCoordinatePrefixes(ResolveStructure, file, extInfo, RetrieveSexOnClothes(file.clothes));
+#else
```
Comment (ManlyMarco) [inline src/Core_Sideloader/UniversalAutoResolver/Core.UAR.Hooks.cs:263]: Why not move the RetrieveSexOnClothes call inside IterateCoordinatePrefixes so all these #ifs are not needed?

File: src/Core_Sideloader/UniversalAutoResolver/Core.UAR.StructReference.AI.cs (hunk 1 above)
Comment (OrangeSpork) [reply src/Core_Sideloader/UniversalAutoResolver/Core.UAR.StructReference.AI.cs:282]: Yes, there's one spot where they do concatenated groupings of both sexes. I followed the same pattern as the character customization for that.

File: src/Core_Sideloader/UniversalAutoResolver/Core.UAR.Hooks.cs (hunk 2 above)
Comment (OrangeSpork) [reply src/Core_Sideloader/UniversalAutoResolver/Core.UAR.Hooks.cs:263]: That can be done...

**Pull request merged**